src/shared/bootspec.c

```diff
@@ -584,6 +584,7 @@ static int boot_entries_select_default(const BootConfig *config) {
         int i;
 
         assert(config);
+        assert(config->entries || config->n_entries == 0);
```

Comment:
so, coverity actually found a bigger issue here... we iterate through the entries with a loop starting with `config->n_entries-1`. Since `n_entries` is `size_t` (i.e. an unsigned type) subtracting one of that will underflow if there are no entries. This is then assigned to an `int` (i.e. a signed type), thus something very high will likely become very low again, which is really weird and confusing. We should probably take more care of the types here, and in fact skip all those three loops if n_entries is zero anyway.

Reply:
I thought about this, but decided to keep the changes minimal, because the code *is* correct. But OK, I'll rework it.